Lib/test/test_dbm_gnu.py

```diff
@@ -144,6 +144,13 @@ def test_nonascii_filename(self):
             self.assertTrue(b'key' in db)
             self.assertEqual(db[b'key'], b'value')
 
+    def test_nonexising_file(self):
+        unlink(filename)
+
+        with self.assertRaisesRegexp(gdbm.error,
+            "\\[Errno 2\\] No such file or directory: '{}'".format(filename)):
```

Comment:
It is better to use raw string literals for regexes.

Since filename can contain regex metacharacters, it should be wrapped with `re.escape()` if used in a regex pattern.

Errno 2 is platform depending.

Is the filename attribute of the exception set?
```python
with self.assertRaises(gdbm.error) as cm:
    gdbm.open(filename)
self.assertIn(filename, str(cm.exception))
self.assertEqual(cm.exception.filename, filename)  # is this true?
```